sdk/core/src/main/java/com/klaviyo/core/config/KlaviyoConfig.kt

```diff
@@ -116,11 +129,41 @@ object KlaviyoConfig : Config {
         private set
     override val networkJitterRange = 0..10
 
-    override fun getManifestInt(key: String, defaultValue: Int): Int = if (!this::applicationContext.isInitialized) {
-        defaultValue
-    } else {
-        applicationContext.getManifestInt(key, defaultValue)
-    }
+    override fun getManifestInt(key: String, defaultValue: Int): Int =
+        if (!this::applicationContext.isInitialized) {
+            defaultValue
+        } else {
+            applicationContext.getManifestInt(key, defaultValue)
+        }
+
+    /**
+     * Helper to grab string resource from application using SDK - uses reflection
+     */
+    @SuppressLint("DiscouragedApi")
+    private fun getReactNativeSdkProperty(key: String): String? =
```

Comment:
not really a fan of how we're doing react native specific stuff, wondering if we can declare resource values in this android sdk side and in the react native sdk, override it using the same key

Reply:
and im wondering what level of validation we need to do since technically, developers would also be able to override these values

Reply:
I can remove the react-native specific naming here for sure. And yeah I agree it's a bit concerning that we're reflecting a resource value that could be overidden - in my research there's not many other ways we can get the value since the dependency route is Android SDK -> React Native SDK -> Client App. I tried using Manifest values but was unsuccessful since we can only attach meta data to an application (which we don't have in the react native SDK) and gradle BuildConfig values cannot be read from the depending sdk. If its any consolation, I did try and override on the Android test app and was not able to, which maybe is a sign that this would only be an issue for react-native clients who add this string resource to their android built folder. 

I like the idea of adding some backend checks around this so we only add config values for strings we expect (ex. android, react-native) so avoid any malicious play here and default to platform if we see anything funky. Would be cool to get another backend ticket in as well, definitely trying to learn more there. For now, I'll update this to be less react-native specific and maybe we can discuss post-hackathon?

Reply:
if we add `res/values/strings.xml` to the android sdk with:
```
<resources>
    <string name="klaviyo_sdk_name">android</string>
</resources>
```
and in the react native side declare the same thing, that'll override it right?

then in the android sdk side, theres no need for reflection, it already knows that R.string.klaviyo_sdk_name should exist